src/viewer.js

```diff
@@ -3990,8 +3956,91 @@ function doViewerResize(viewer, containerSize){
     THIS[viewer.hash].needsResize = false;
     THIS[viewer.hash].forceResize = false;
 }
+
+function handleArrowKeys(viewer) {
+
+    // Helper for key state
+    function isDown(code) {
+        return (viewer._keysDown && viewer._keysDown[code]) || (viewer._keyVirtuallyHeld && viewer._keyVirtuallyHeld[code]);
+    }
+
+    // Helper to process pan keys: handles pan, pan distance, and virtual hold logic
+    function handlePan(codeList, deltaPoint) {
+        for (const code of codeList) {
+            if (isDown(code)) {
+                viewer.viewport.panBy(deltaPoint);
+                viewer.viewport.applyConstraints();
+                viewer._keyPanDistance[code] += viewer.pixelsPerArrowPress;
+                if (viewer._keyVirtuallyHeld && viewer._keyVirtuallyHeld[code] && viewer._keyPanDistance[code] >= viewer.minPanDistance) {
+                    viewer._keyVirtuallyHeld[code] = false;
+                    viewer._keyPanDistance[code] = 0;
+                }
+                break; // Only pan once per group
+            }
+        }
+    }
+
+    // Helper to process zoom keys
+    function handleZoom(codeList, factor) {
+        for (const code of codeList) {
+            if (isDown(code)) {
+                viewer.viewport.zoomBy(factor);
+                viewer.viewport.applyConstraints();
+                break;
+            }
+        }
+    }
+
+    // Use the viewer's configured pan amount
+    const pixels = viewer.pixelsPerArrowPress;
```

Comment:
We need to adjust this to compensate for the fact we are now doing it every frame instead of every key press. Dividing by 10 seems good.